client/landing/stepper/hooks/use-anchor-parameter.ts

```diff
@@ -0,0 +1,50 @@
+import { useLocation } from 'react-router-dom';
+
+export interface AnchorStateType {
+	anchorFmPodcastId?: string;
+	anchorFmEpisodeId?: string;
+	anchorFmSpotifyUrl?: string;
+	anchorFmSite?: string;
+	anchorFmPost?: string;
+	anchorFmIsNewSite?: string;
+}
+export type AnchorStateKeyType = keyof AnchorStateType;
+
+/*
+ useAnchorParameter is an internal helper for finding a value that comes from either a query string, or location state.
+ For example, when a user hits http://calypso.localhost:3000/new?anchor_podcast=40a166a8
+ We grab that anchor_podcast value, and store it in location state to keep as we move along the states.
+ It's called "anchor_podcast" in the query string above, but "anchorFmPodcastId" above.
+  Calling this function like so:
+  useAnchorParameter({
+    queryParamName: 'anchor_podcast',
+    locationStateParamName: 'anchorFmPodcastId',
+  })
+  Looks for the value first in location state, then if it can't be found, looks in the query parameter.
+*/
+interface UseAnchorParameterType {
+	queryParamName: string;
+	locationStateParamName: AnchorStateKeyType;
+	sanitize: ( arg0: string ) => string;
+}
+
+export default function useAnchorParameter( {
```

Comment:
Discussion point, not necessarily a critique - would it perhaps be a good idea to make this a more generic "location state with fallback to query param" hook and not restricted to anchor?

Reply:
This makes good sense to me in the event we need to use the same logic for other query params. I'll update it to `useParameter` so it's not specific to Anchor. :)